middlewares/authenticator_test.go

```diff
@@ -5,13 +5,52 @@ import (
 	"io/ioutil"
 	"net/http"
 	"net/http/httptest"
+	"os"
 	"testing"
 
 	"github.com/codegangsta/negroni"
 	"github.com/containous/traefik/types"
 	"github.com/stretchr/testify/assert"
 )
 
+func TestBasicAuthUsersFromFile(t *testing.T) {
+	usersStr := "test:$apr1$H6uskkkW$IgXLP6ewTrSuBkTrqE8wj/\ntest2:$apr1$d9hr9HBB$4HxwgUir3HP4EsggP/QNo0\n"
+	usersFile, err := ioutil.TempFile("", "basic-users")
+	assert.NoError(t, err, "there should be no error")
+	defer os.Remove(usersFile.Name())
+	_, err = usersFile.Write([]byte(usersStr))
+	assert.NoError(t, err, "there should be no error")
+	basic := &types.Basic{
+		UsersFile: usersFile.Name(),
+	}
+	users, err := parserBasicUsers(basic)
+	assert.NoError(t, err, "there should be no error")
+	assert.Equal(t, 2, len(users), "they should be equal")
+	_, ok := users["test"]
+	assert.True(t, ok, "user test should be found")
+	_, ok = users["test2"]
+	assert.True(t, ok, "user test2 should be found")
+}
+
+func TestDigestAuthUsersFromFile(t *testing.T) {
+	usersStr := "test:traefik:a2688e031edb4be6a3797f3882655c05 \ntest2:traefik:518845800f9e2bfb1f1f740ec24f074e\n"
+	usersFile, err := ioutil.TempFile("", "basic-users")
+	assert.NoError(t, err, "there should be no error")
+	defer os.Remove(usersFile.Name())
+	_, err = usersFile.Write([]byte(usersStr))
+	assert.NoError(t, err, "there should be no error")
+	digest := &types.Digest{
+		UsersFile: usersFile.Name(),
+	}
+	users, err := parserDigestUsers(digest)
+	assert.NoError(t, err, "there should be no error")
+	assert.Equal(t, 2, len(users), "they should be equal")
+	_, ok := users["test:traefik"]
+	assert.True(t, ok, "user test should be found")
+	_, ok = users["test:traefik"]
+	assert.True(t, ok, "user test2 should be found")
+}
```

Comment:
Wouldn't a table-driven test make sense when the all the inputs (in a given column) are are of the same? Or when the function under test is the same for each row?

In these two tests, the fixtures (inputs) are different types and the functions under test are distinct. i.e. I humbly suggest the similarity between these two tests is on a superficial level and trying to combine  these will not be trivial... if accomplished, it's certain to obfuscate what is otherwise some very simple, straightforward test logic.